weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ThreadsPage/ThreadsPage.tsx

```diff
@@ -44,34 +103,210 @@ export const ThreadsPage: FC<{
   paginationModel: GridPaginationModel;
   setPaginationModel: (newModel: GridPaginationModel) => void;
 }> = props => {
-  const {baseView, views} = props;
+  const {entity, project} = useEntityProject();
+  const {baseView, views, fetchViews, onRecordLastView} = props;
 
   const table = 'threads';
   const view = baseView.object_id;
 
+  const getTsClient = useGetTraceServerClientContext();
+  const tsClient = getTsClient();
+  const currentViewDefinition = useCurrentViewDefinitionForThreads(
+    baseView,
+    props.filterModel,
+    props.sortModel,
+    props.paginationModel,
+    props.columnVisibilityModel,
+    props.pinModel
+  );
+  const history = useHistory();
+
+  const createSavedView = useCreateSavedView(entity, project, table);
+
+  const [upsertState, onUpsertView] = useAsyncFn(
+    async (
+      objectId: string,
+      label: string | null,
+      successMessage: string,
+      reloadAfter: boolean
+    ) => {
+      try {
+        const resolvedLabel = label ?? baseView.val.label;
+        await createSavedView(objectId, resolvedLabel, currentViewDefinition);
+
+        const newQuery = new URLSearchParams(history.location.search);
+        newQuery.set('view', objectId);
+        history.push({search: newQuery.toString()});
+
+        if (reloadAfter) {
+          fetchViews();
+        }
+
+        toast(successMessage);
+        onRecordLastView(objectId);
+      } catch (error) {
+        toast('Failed to persist Saved View.', {
+          type: 'error',
+        });
+        throw error; // Re-throw to maintain useAsyncFn error state if needed
+      }
+    },
+    [
+      baseView.val.label,
+      createSavedView,
+      currentViewDefinition,
+      fetchViews,
+      history,
+      onRecordLastView,
+    ]
+  );
+
+  const onLoadView = useCallback(
+    (viewToLoad: TraceObjSchema) => {
+      // We want to preserve any params that are not part of view definition,
+      // e.g. peek drawer state.
+      const newQuery = new URLSearchParams(history.location.search);
+
+      // Clear out any params related to saved views
+      // For threads, we only track filters, sort, pageSize
```

Comment:
I think this comment is out of date now (cols and pins) I'd probably just remove it.